types/types.go

```diff
@@ -56,12 +56,15 @@ type Image interface {
 // ImageManifest is the interesting subset of metadata about an Image.
 // TODO(runcom)
 type ImageManifest interface {
+	Raw() []byte
+	Layers() []string
 	String() string
 }
 
 // DockerImageManifest is a set of metadata describing Docker images and their manifest.json files.
 // Note that this is not exactly manifest.json, e.g. some fields have been added.
 type DockerImageManifest struct {
```

Comment:
Quite a few fields have been added; most worrisome to me are those (e.g. `Name`, `Tag`, `RepoTags`) which can’t be parsed from the manifest and are not a property of the manifest. It works, of course, but it feels unclean.